modules/planning/common/open_space_info.h

```diff
@@ -334,6 +350,8 @@ class OpenSpaceInfo {
 
   std::pair<PublishableTrajectory, canbus::Chassis::GearPosition>
       publishable_trajectory_data_;
+
+  apollo::planning_internal::Debug* debug_;
```

Comment:
The ownership is unclear and dangerous. Use shared_ptr if you would like, though better to redesign.

Reply:
> The ownership is unclear and dangerous. Use shared_ptr if you would like, though better to redesign.

The debug pointer is passed by ADCtrajectory and we will modify the structure later.

Reply:
Not good for ownership and lifecycle management. You can just use the debug_instance_ and copy to the external ADCtrajectory in the end. Cleaner and safer.